webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java

```diff
@@ -664,6 +665,96 @@ private static void addFmcSolutionSheet(PdfWriter docWriter, Document doc, Scram
+
+        // Face moves.
+        for (int i=0; i<directionModifiers.length; i++){
+            for (int j=0; j<moves.length; j++){
+                movesCell[0][i][j] = moves[j]+directionModifiers[i];
+            }
+        }
+        // Rotations.
+        for (int i=0; i<directionModifiers.length; i++){
+            for (int j=0; j<moves.length; j++){
+                movesCell[1][i][j] = "["+moves[j].toLowerCase()+directionModifiers[i]+"]";
+            }
+        }
+
+        float maxFirstColumnWidth = 0; // Variable used to center the table.
+
+        for (int i=0; i<movesType.length; i++) {
+            Rectangle currentRectangle = new Rectangle(firstColumnWidth, tableHeight);
+
+            Font tempFont = new Font(bf);
+            float size = fitText(tempFont, movesType[i], currentRectangle, 10, false, 1f);
+
+            maxFirstColumnWidth = Math.max(maxFirstColumnWidth, tempFont.getBaseFont().getWidthPoint(movesType[i], size));
+
+            PdfPCell cell = new PdfPCell(new Phrase(movesType[i], new Font(bf, size, Font.BOLD)));
+            cell.setFixedHeight(cellHeight);
+            cell.setVerticalAlignment(Element.ALIGN_MIDDLE);
+            cell.setHorizontalAlignment(Element.ALIGN_RIGHT);
+            cell.setBorder(Rectangle.NO_BORDER);
+            table.addCell(cell);
+
+            cell = new PdfPCell(new Phrase(""));
+            cell.setFixedHeight(cellHeight);
+            cell.setColspan(columns-1);
+            cell.setBorder(Rectangle.NO_BORDER);
+            table.addCell(cell);
+
+            for (int j=0; j<directionModifiers.length; j++) {
+                size = fitText(new Font(bf), direction[j], currentRectangle, 10, false, 1f);
+
+                maxFirstColumnWidth = Math.max(maxFirstColumnWidth, tempFont.getBaseFont().getWidthPoint(direction[j], size));
+
+                cell = new PdfPCell(new Phrase(direction[j], new Font(bf, size)));
+                cell.setFixedHeight(cellHeight);
+                cell.setVerticalAlignment(Element.ALIGN_MIDDLE);
+                cell.setHorizontalAlignment(Element.ALIGN_RIGHT);
+                cell.setBorder(Rectangle.NO_BORDER);
+                table.addCell(cell);
+                for (int k=0; k<moves.length; k++) {
+                    cell = new PdfPCell(new Phrase(movesCell[i][j][k], new Font(bf, 10)));
+                    cell.setFixedHeight(cellHeight);
+                    cell.setVerticalAlignment(Element.ALIGN_MIDDLE);
+                    cell.setHorizontalAlignment(Element.ALIGN_CENTER);
+                    cell.setBorder(Rectangle.NO_BORDER);
+                    table.addCell(cell);
+                }
+            }
+        }
+
+        // Center
+        float tableOffset = (tableWidth-(columns-1)*cellWidth-maxFirstColumnWidth)/2;
+
+        // Position the table
+        table.writeSelectedRows(0, -1, left-tableOffset, scrambleBorderTop+tableHeight, cb);
+
+        // Rules
```

Comment:
Is this the update? I have just copy-pasted from the original code to revert the gray stuff and...